src/source/Signaling/Client.c

```diff
@@ -16,17 +71,45 @@ STATUS createSignalingClientSync(PSignalingClientInfo pClientInfo, PChannelInfo
     MEMSET(&signalingClientInfoInternal, 0x00, SIZEOF(signalingClientInfoInternal));
     signalingClientInfoInternal.signalingClientInfo = *pClientInfo;
 
-    CHK_STATUS(createSignalingSync(&signalingClientInfoInternal, pChannelInfo, pCallbacks, pCredentialProvider, &pSignalingClient));
+    CHK_STATUS(createRetryStrategyForCreatingSignalingClient(pClientInfo, &createSignalingClientRetryStrategy));
 
-    *pSignalingHandle = TO_SIGNALING_CLIENT_HANDLE(pSignalingClient);
+    signalingClientCreationMaxRetryCount = pClientInfo->signalingClientCreationMaxRetryAttempts;
+    while (TRUE) {
```

Comment:
Why are we doing a while(TRUE) and then checking for `signalingClientCreationMaxRetryCount` inside? Directly looping on the variable would avoid using break altogether.

Reply:
Ok. I get it. We want to make sure we dont wait after the last attempt. 
So, with, `while(signalingClientCreationMaxRetryCount >= 0)`, if `signalingClientCreationMaxRetryCount` fails, we would end up invoking exponentialBackOff unnecessarily after the last try. With this, we avoid doing that.